stock_reserve_sale/model/sale.py

```diff
@@ -0,0 +1,160 @@
+    def _compute_stock_reservation(self):
+        for sale in self:
+            has_stock_reservation = False
+            is_stock_reservable = False
+            for line in sale.order_line:
+                if line.reservation_ids:
+                    has_stock_reservation = True
+                if line.is_stock_reservable:
+                    is_stock_reservable = True
+            if sale.state not in ("draft", "sent"):
+                is_stock_reservable = False
+            sale.is_stock_reservable = is_stock_reservable
+            sale.has_stock_reservation = has_stock_reservation
+
+    has_stock_reservation = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        multi="stock_reservation",
+        store=True,
+        string="Has Stock Reservations",
+    )
+    is_stock_reservable = fields.Boolean(
+        compute="_compute_stock_reservation",
+        readonly=True,
+        multi="stock_reservation",
+        store=True,
+        string="Can Have Stock Reservations",
+    )
+    reserves_count = fields.Integer(compute="_compute_reserves_count")
+    all_lines_reserved = fields.Boolean(
+        compute="_compute_reserves_count", store=True, default=False
+    )
+
+    def release_all_stock_reservation(self):
+        line_ids = [line.id for order in self for line in order.order_line]
+        lines = self.order_line.browse(line_ids)
+        lines.release_stock_reservation()
+        return True
+
+    def _compute_reserves_count(self):
+        reserve_ids = self.env["stock.reservation"]._read_group(
+            domain=expression.AND(
+                [
+                    [("sale_id", "in", self.ids)],
+                ]
+            ),
+            fields=["sale_id"],
+            groupby=["sale_id"],
+        )
+        lines = self.order_line.filtered(
+            lambda l: l.product_id.detailed_type != "service"
+        )
+        for order in self:
+            if reserve_ids:
+                order.reserves_count = reserve_ids[0]["sale_id_count"]
+            else:
+                order.reserves_count = 0
+            if order.reserves_count == len(lines):
+                order.all_lines_reserved = True
+            else:
+                order.all_lines_reserved = False
+
+    def action_view_reserves_products(self):
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "stock_reserve.action_stock_reservation_tree"
+        )
+        action["domain"] = [("sale_id", "in", self.ids)]
+        action["context"] = {"search_default_groupby_product": False}
+        return action
+
+    def action_view_reserves_stock_picking(self):
+        stock_picking = ""
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "stock.action_picking_tree_all"
+        )
+        stock_picking = (
+            self.env["stock.picking"]
+            .search([("origin", "=", self.name)])
+            .filtered(lambda a: a.state not in "cancel")
```

Comment:
This is not correct at all.
The value of state could be "draft", "cancel",...

As is, this mean:
`"draft" in ("c", "a", "n", "c", "e", "l")` ? => `True/False`

It should be:
```
# Use limit=1 because we use the record with .id later
stock_picking = self.env["stock.picking"].search([("origin", "=", self.name), ("state", "!=", "cancel")], limit=1)
```